src/main/java/login/UserManager.java

```diff
@@ -1,41 +1,129 @@
+            LoginOutputBound.CreatAccountSuccess();
+        }else{LoginOutputBound.CreatAccountFail();}
     }
 
 
-    // method to add user to the allUsers when user created the account,need to check if the username already in the list
 
     // method to find the user in the allUsers and change the status to logged in
+    public void Login(String username, String password){
+        for(User person :AllUser.getAllUser()){
+            if(Objects.equals(person.getUsername(), username) && Objects.equals(person.getPassword(), password)){
+                person.setLoginStatus(true);
+                LoginOutputBound.LoginSuccess();
+            }else{
+                LoginOutputBound.LoginFailed();
+                }
+
+            }
+        }
+
+
     // method to find the user in the allUsers and change the status to logged out
+    public void Logout(String username){
+        for(User person :AllUser.getAllUser()){
+            if(Objects.equals(person.getUsername(), username)){
+                person.setLoginStatus(false);
+            }else{LoginOutputBound.LogoutFail();}
+
+        }
+
+
+    }
+    // method to find the user in the allUsers and add profile
+
+    public void AddProfile(String username, Profile profile){
+        for(User person :AllUser.getAllUser()){
+            if(Objects.equals(person.getUsername(), username)){
+                person.setProfile(profile);}else{LoginOutputBound.AddProfile();}
+
+        }
+
+    }
 
+    //Check if user already followed this other user(refactoring)
+    public boolean CheckFollow(String username, User other){
+        for(User person :AllUser.getAllUser()){
+            if(Objects.equals(person.getUsername(), username)){
+                if (person.getFollowed().contains(other)){
+                    LoginOutputBound.FollowedFail();
+                    return false;
+                }
+            }
+            }return true;
+
+        }
+
+    //a user want to follow another user
+    // add other to user.followed
+    //other.follower add user
     // method to find the user in the allUsers and add the other user in to the followed list(attribute of user),need to
     //check if the other user is already followed,also find the other user and add the user from the following list(attribute of user)
+    public void follow(String username, User other){
+        for(User person :AllUser.getAllUser()){
+            if(Objects.equals(person.getUsername(), username)){
+                person.addFollowed(other);
+                other.addFollowers(person);
+                LoginOutputBound.FollowedSuccess();
+            }
+
+        }
+
+
+    }
 
     // method to find the user in the allUsers and remove the other user in to the followed list(attribute of user),need to
     //check if the other user is already followed,also find the other user and remove the user from the following list(attribute of user)
+    public boolean CheckUnFollow(String username, User other){
```

Comment:
Does the use case have to check every time? Maybe it's simpler just to surround unfollow in a try/catch? I am not sure but feel free to explore!

Reply:
I think it should check every time when user tries to unfollow another user, If the user tries to unfollow this same user many times(if thats what you mean), the presenter will tell the user that he/she has already unfollowed that user.